drivers/rtc/rtc_ll_stm32.c

```diff
@@ -259,6 +259,15 @@ static int rtc_stm32_get_time(const struct device *dev, struct rtc_time *timeptr
 		return err;
 	}
 
+	if (!LL_RTC_IsActiveFlag_INITS(RTC)) {
+		/* INITS flag is set when the calendar has been initialiazed. This flag is
+		 * reset only on backup domain reset, so it can be read after a system
+		 * reset to check if the calendar has been initialized.
+		 */
+		k_mutex_unlock(&data->lock);
```

Comment:
Is a k_mutex_lock/k_mutex_unlock   required for checking the INITS bit ?

Reply:
According to https://github.com/zephyrproject-rtos/zephyr/commit/a8b7076099478b6b8e4db72fa86fb0ad1edbaf72:
> This adds mutex a lock in rtc_stm32_set_time and rtc_stm32_get_time to
> ensure consistent data reading. 

I suppose it would take setting BDRST in RCC_BDCR for the RTC to go back to the uninitialized state. I did not see it used in the RTC implementation, so the mutex is probably not necessary to ensure consistent data reading.

Also, if I read the documentation literally:
> Bit 4 INITS: Initialization status flag
> This bit is set by hardware when the calendar year field is different from 0 (Backup domain
> reset state).
> 0: Calendar has not been initialized
> 1: Calendar has been initialized

It means that the year 00 will cause INITS to return 0. This means that the RTC cannot represent past time (year 2000) and will stop working in year 2100.